### PR TITLE
[Task #3] displaying recipe and makes a dish ready

### PR DESCRIPTION
**Pull request by Aleksander-Drumev-WA**

### CODE REVIEW

File: Commands/CompleteDishesCommand.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        {
+            _restaurant = restaurant;
+            _loadOrdersCommand = loadOrdersCommand;
+        }
+
+        public override async Task ExecuteAsync(object? parameter)
+        {
+            try
+            {
+                var values = (object[])parameter;
```
Comment (StanYFed) [inline Commands/CompleteDishesCommand.cs:29]: Lets make this code more safety:
```
if (parameter is object[] values && 
                    values[0] is int dishId &&
                    values[1] is int orderNumber) {
```

File: Commands/ShowRecipeCommand.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        private MainChefViewModel _mainChefViewModel;
+        private readonly Restaurant _restaurant;
+
+        public ShowRecipeCommand(MainChefViewModel mainChefViewModel, Restaurant restaurant)
+        {
+            _mainChefViewModel = mainChefViewModel;
+            _restaurant = restaurant;
+        }
+
+        public override void Execute(object? parameter)
```
Comment (StanYFed) [inline Commands/ShowRecipeCommand.cs:26]: Why we send array here? Maybe we can send whole ViewModel, see other comment start with "If we bind whole ViewModel"

File: Views/UserControls/ChefLookingAtRecipeView.xaml [hunk 3]
```diff
@@ -35,6 +39,14 @@ Frost. Remove from the oven and allow to cool for about 10 minutes, remove from
                 HorizontalAlignment="Right"
                 Margin="0 0 15 10"
                 Padding="35 8"
-                FontSize="14"/>
+                FontSize="14"
+                Command="{Binding CompleteDishesCommand}">
+            <Button.CommandParameter>
```
Comment (StanYFed) [inline Views/UserControls/ChefLookingAtRecipeView.xaml:44]: If we bind whole ViewModel we avoid MyMultiValueConverter
`
CommandParameter ="{Binding}"
`

File: Commands/ShowRecipeCommand.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        {
+            try
+            {
+                var values = (object[])parameter;
+
+                var dishId = (int)values[0];
+                var orderId = (int)values[1];
+
+                var chosenDish = _mainChefViewModel.Orders.SelectMany(x => x.OrderItems.Where(oi => oi.Id == dishId && oi.OrderNumber == orderId)).FirstOrDefault();
```
Comment (StanYFed) [inline Commands/ShowRecipeCommand.cs:35]: We have orderId and dishId, so
1. Filter orders `_mainChefViewModel.Orders.SignleOrDefault(....)`
2. Then dishes in order `?.OrderItems.FirstOrDefault(...)`
Correct?

File: Resources/Converters/BooleanToVisibilityConverter.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+            }
+            else
+            {
+                return Visibility.Hidden;
+            }
+        }
+
+        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            return Binding.DoNothing;
```
Comment (StanYFed) [inline Resources/Converters/BooleanToVisibilityConverter.cs:29]: О! I didn't know about this! Cool!

File: Services/Data/Providers/DatabaseOrdersProvider.cs [hunk 6]
```diff
@@ -26,13 +26,27 @@ public async Task<IEnumerable<Order>> GetAllOrders()
...
         }
+
+        public async Task CompleteDish(int dishId, int orderNumber)
+        {
+            using (var dbContext = _dbContextFactory.CreateDbContext())
+            {
+                var dishesToChangeCompletedState = await dbContext.OrderItems.Where(oi => oi.DishId == dishId && oi.OrderId == orderNumber).ToListAsync();
+
+
+                dishesToChangeCompletedState.ForEach(oi => oi.IsCompleted = true);
```
Comment (StanYFed) [inline Services/Data/Providers/DatabaseOrdersProvider.cs:45]: It's not clear from the task, but the dishes are marked ready one at a time.
I'll fix it in the task.

**Review by StanYFed — CHANGES_REQUESTED** [review]

File: Commands/ShowRecipeCommand.cs (hunk 4 above)
Comment (Aleksander-Drumev-WA) [reply Commands/ShowRecipeCommand.cs:35]: `chosenDish` variable is getting the right dish from the list of orders

**Pull request merged**